bigbluebutton-html5/imports/ui/components/chat/message-list/message-list-item/component.jsx

```diff
@@ -160,6 +159,100 @@ class MessageListItem extends Component {
       </div>
     );
   }
+
+  renderPollItem(pollMessage) {
+    const {
+      user,
+      time,
+      intl,
+    } = this.props;
+
+    const dateTime = new Date(time);
+
+    let pollText = [];
+    const pollElement = [];
+
+    pollMessage.forEach((msg) => {
+      pollText = msg.text.split('<br/>');
+      pollElement.push(pollText.map(o => <div key={_.uniqueId('chat-poll-result-')}>{o}</div>));
+
+      if (pollMessage.length > 1) {
+        pollElement.push(
+          <div key={_.uniqueId('chat-poll-separator-')} className={styles.divider} />,
+        );
+      }
+    });
+
+    return pollMessage ? (
+      <div className={styles.item} key={_.uniqueId('message-poll-item-')}>
+        <div className={styles.wrapper} ref={(ref) => { this.item = ref; }}>
+          <div className={styles.avatarWrapper}>
+            <UserAvatar
+              className={styles.avatar}
+              color={user.color}
+              moderator={user.isModerator}
+            >
+              {<Icon className={styles.isPoll} iconName="polling" />}
+            </UserAvatar>
+          </div>
+          <div className={styles.content}>
+            <div className={styles.meta}>
+              <div className={styles.name}>
+                <span>{intl.formatMessage(intlMessages.pollResult)}</span>
+              </div>
+              <time className={styles.time} dateTime={dateTime}>
+                <FormattedTime value={dateTime} />
+              </time>
+            </div>
+            <div className={styles.messages}>
+              {pollMessage[0] ? (
+                <div>
+                  {
+                  pollElement
+                }
+                </div>
+              ) : null}
+            </div>
+          </div>
+        </div>
+      </div>
+    ) : null;
+  }
+
+  render() {
+    const {
+      user,
+      messages,
+    } = this.props;
+
+    if (!user) {
+      return this.renderSystemMessage();
+    }
+
+    const chatMessages = [];
+    const pollMessage = [];
+
+    if (messages.length > 0) {
+      messages.forEach((message) => {
```

Comment:
I don't think pulling *all* chat messages and iterating over them inside of a frequently called method (render) is a good idea... At the very least separate the chat from polling before rendering, in a container/service somewhere earlier where it does not get triggered often. Ideally only once...

Reply:
The original `messages` was correct. Please put it back